Mixpanel/Mixpanel.m

```diff
@@ -305,8 +310,16 @@ - (void)createAlias:(NSString *)alias forDistinctID:(NSString *)distinctID
         MPLogError(@"%@ create alias called with empty distinct id: %@", self, distinctID);
         return;
     }
-    [self track:@"$create_alias" properties:@{ @"distinct_id": distinctID, @"alias": alias }];
-    [self flush];
+    if (alias != distinctID) {
+        self.alias = alias;
```

Comment:
i think setting this should be in the same `dispatch_async` block as `archiveProperties`. whether that's before or after `track` is up to you, but that way it'll be set and archived and tracked all in sequence instead of being set potentially way before the rest depending on how much is in the queue at the time